src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1400,6 +1401,37 @@ internal override bool HasDeclaredRequiredMembers
             }
         }
 
+        internal override bool HasAsyncMethodBuilderAttribute(out TypeSymbol? builderArgument)
+        {
+            return HasAsyncMethodBuilderAttribute(this, out builderArgument);
+        }
+
+        /// <summary>
+        /// Returns true if the method has a [AsyncMethodBuilder(typeof(B))] attribute. If so it returns type B.
+        /// Validation of builder type B is left for elsewhere. This method returns B without validation of any kind.
+        /// </summary>
+        internal static bool HasAsyncMethodBuilderAttribute(Symbol symbol, [NotNullWhen(true)] out TypeSymbol? builderArgument)
+        {
+            Debug.Assert(symbol is not null);
+
+            // Find the AsyncMethodBuilder attribute.
+            foreach (var attr in symbol.GetAttributes())
+            {
+                Debug.Assert(attr is SourceAttributeData);
+
+                if (attr.IsTargetAttribute(AttributeDescription.AsyncMethodBuilderAttribute)
```

Comment:
I thought you had a PR which also changes this kind of pattern to check "GetSignatureIndex" instead. Is there any motivation to also make such a change here?

Reply:
> Is there any motivation to also make such a change here?

This is a general pattern that we use for detecting well-known attributes in source when there is only one signature that can match. I think it is robust enough for attributes from source. In other PRs I was mostly adjusting code that was supposed to handle attributes from metadata as well. In those cases, if there was a mismatch between the signature and the actual arguments, my goal was to report an error rather than simply skip the attribute. Here, there is no intent to do that because errors are reported during binding the attribute.